core/watcher/watchertest/watcher.go

```diff
@@ -44,3 +47,49 @@ func DirtyKill[T any](c *gc.C, w watcher.Watcher[T]) {
 		c.Logf("ignoring failed to close for watcher")
 	}
 }
+
+// WatcherC embeds a gocheck.C and adds methods to help
+// verify the behaviour of generic watchers.
+type WatcherC[T any] struct {
+	C       *gc.C
+	Watcher watcher.Watcher[T]
+}
+
+// NewWatcherC() returns a WatcherC[T].
+func NewWatcherC[T any](c *gc.C, w watcher.Watcher[T]) WatcherC[T] {
+	return WatcherC[T]{
+		C:       c,
+		Watcher: w,
+	}
+}
+
+// AssertNoChange verifies that no changes are received from the watcher during
+// a testing.ShortWait time.
+func (w *WatcherC[T]) AssertNoChange() {
```

Comment:
I'm a bit ambivalent on using this. 

The deterministic pattern (and one that doesn't involve arbitrary waits), is to cause state side-effects via specific service methods, see the watcher changes you want, then assert that the _change stream_ is idle.

We know that the change-stream is idle because it does a check, sees nothing to dispatch and declares itself so. This is working with discrete, provable behaviours.